flair/datasets/ocr.py

```diff
@@ -0,0 +1,250 @@
+import json
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+import gdown.download_folder
+import PIL
+from torch.utils.data import Dataset
+
+import flair
+from flair.data import BoundingBox, Corpus, FlairDataset, Sentence
+from flair.datasets.base import find_train_dev_test_files
+from flair.models.sequence_tagger_utils.bioes import get_spans_from_bio
+
+
+class OcrJsonDataset(FlairDataset):
+    def __init__(
+        self,
+        path_to_split_directory: Union[str, Path],
+        label_type: str = "ner",
+        in_memory: bool = True,
+        encoding: str = "utf-8",
+        load_images: bool = False,
+        normalize_coords_to_thousands: bool = True,
+        label_name_map: Dict[str, str] = None,
+    ):
+        """
+        Instantiates a Dataset from a OCR-Json format.
+        The folder is structured with a "images" folder and a "tagged" folder.
+        Those folders contain respectively .jpg an .json files with matching file name.
+        The json contains 3 fields "words", "bbox", "labels" which are lists of equal length
+        "words" is a list of strings, containing the ocr texts,
+        "bbox" is a list of int-Tuples, containing left, top, right, bottom
+        "labels" is a BIO-tagging of the sentences
+        :param path_to_split_directory: base folder with the task data
+        :param label_type: the label_type to add the ocr labels to
+        :param encoding: the encoding to load the .json files with
+        :param normalize_coords_to_thousands: if True, the coordinates will be ranged from 0 to 1000
```

Comment:
is normalizing to thousands usual? If it is just selected by random, why not make the normalization factor as int and optional and normalize images if factor is provided.

Reply:
Normalizing to thousands is very usual, it is done by Layoutlm (& v2/v3), Docformer, Lambert, ...
I haven't seen an implementation so far that did it different